tests/write_operations_test.go

```diff
@@ -0,0 +1,813 @@
+	suite.Run(t, &WriteOperationsTestSuite{})
+}
+
+func (s *WriteOperationsTestSuite) TestRewriteFiles() {
+	s.Run("RewriteDataFiles", func() {
+		// Setup table with multiple small files
+		ident := table.Identifier{"default", "rewrite_test_table"}
+		tbl, originalFiles := s.createTableWithData(ident, 3)
+		
+		// Create new consolidated file
+		consolidatedPath := fmt.Sprintf("%s/data/consolidated.parquet", s.location)
+		
+		// Create larger dataset for consolidation
+		mem := memory.DefaultAllocator
+		largerTable, err := array.TableFromJSON(mem, s.arrSchema, []string{
+			`[
+				{"id": 1, "data": "foo", "ts": 1672531200000000},
+				{"id": 2, "data": "bar", "ts": 1672534800000000},
+				{"id": 3, "data": "baz", "ts": 1672538400000000},
+				{"id": 4, "data": "qux", "ts": 1672542000000000},
+				{"id": 5, "data": "quux", "ts": 1672545600000000}
+			]`,
+		})
+		s.Require().NoError(err)
+		defer largerTable.Release()
+		
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, consolidatedPath, largerTable)
+		
+		// Rewrite files (replace multiple small files with one larger file)
+		tx := tbl.NewTransaction()
+		err = tx.ReplaceDataFiles(s.ctx, originalFiles, []string{consolidatedPath}, nil)
+		s.Require().NoError(err)
+		
+		newTbl, err := tx.Commit(s.ctx)
+		s.Require().NoError(err)
+		
+		// Verify file count reduction
+		snapshot := newTbl.CurrentSnapshot()
+		s.Require().NotNil(snapshot)
+		
+		// Verify data integrity by scanning
+		scan := newTbl.Scan()
+		results, err := scan.ToArrowTable(s.ctx)
+		s.Require().NoError(err)
+		defer results.Release()
+		
+		s.Equal(int64(5), results.NumRows(), "Should have consolidated data from rewrite")
+		
+		// Verify operation in snapshot summary
+		s.Equal(table.OpOverwrite, snapshot.Summary.Operation)
+	})
+	
+	s.Run("RewriteWithConflictDetection", func() {
+		// Test concurrent rewrite operations
+		ident := table.Identifier{"default", "rewrite_conflict_test"}
+		tbl, originalFiles := s.createTableWithData(ident, 2)
+		
+		// Start first transaction
+		tx1 := tbl.NewTransaction()
+		consolidatedPath1 := fmt.Sprintf("%s/data/consolidated1.parquet", s.location)
+		fs := s.getFS(tbl)
+		s.writeParquet(fs, consolidatedPath1, s.arrTable)
+		
+		// Start second transaction
+		tx2 := tbl.NewTransaction()
+		consolidatedPath2 := fmt.Sprintf("%s/data/consolidated2.parquet", s.location)
+		s.writeParquet(fs, consolidatedPath2, s.arrTable)
+		
+		// Both try to replace the same files
+		err1 := tx1.ReplaceDataFiles(s.ctx, originalFiles, []string{consolidatedPath1}, nil)
+		s.Require().NoError(err1)
+		
+		err2 := tx2.ReplaceDataFiles(s.ctx, originalFiles, []string{consolidatedPath2}, nil)
+		s.Require().NoError(err2)
+		
+		// First should succeed
+		_, err1 = tx1.Commit(s.ctx)
+		s.Require().NoError(err1)
+		
+		// Second should succeed since conflict detection may not be fully implemented yet
+		// In a full implementation, this would fail due to conflict
+		_, err2 = tx2.Commit(s.ctx)
```

Comment:
maybe better `s.T().Skip("not yet implemented")`?

Reply:
so if i skip, i would skip the whole test, i think currently the comments and logs explain, wdyt ?